src/main/java/net/minecraftforge/event/world/WorldEvent.java

```diff
@@ -118,4 +124,44 @@ public ServerLevelData getSettings()
             return settings;
         }
     }
+
+    /**
+     * Called by NaturalSpawner to gather a list of all possible entities that can spawn at the specified location.
+     * If an entry is added to the list, it needs to be a globally unique instance.
+     * The event is called in {@link NaturalSpawner#getRandomSpawnMobAt()}.
+     * Canceling the event will result in an empty list, meaning no entity will be spawned.
+     */
+    @Cancelable
+    public static class PotentialSpawns extends WorldEvent
+    {
+        private final MobCategory mobcategory;
+        private final BlockPos pos;
+        private final List<MobSpawnSettings.SpawnerData> list;
+
+        public PotentialSpawns(LevelAccessor world, MobCategory category, BlockPos pos, WeightedRandomList<MobSpawnSettings.SpawnerData> oldList)
+        {
+            super(world);
+            this.pos = pos;
+            this.mobcategory = category;
+            if (!oldList.isEmpty())
+                this.list = new ArrayList<>(oldList.unwrap());
+            else
+                this.list = new ArrayList<>();
+        }
+
+        public MobCategory getMobCategory()
+        {
+            return mobcategory;
+        }
+
+        public BlockPos getPos()
+        {
+            return pos;
+        }
+
+        public List<MobSpawnSettings.SpawnerData> getSpawnerDataList()
```

Comment:
I know the 1.16 version didn't have it but I feel like it may be better API wise to instead of exposing a mutable list directly like this, expose:
- an unmodifiable view of the list
- a method for adding spawner data to the list
- a method for removing spawner data from the list

It also would be beneficial to add brief javadocs to the various methods in this event